pom.xml

```diff
@@ -848,6 +848,14 @@
         <scope>test</scope>
       </dependency>
 
+      <dependency>
+        <groupId>org.junit</groupId>
+        <artifactId>junit-bom</artifactId>
+        <version>${junit.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
```

Comment:
Any explanation for this?

Reply:
I'm not really sure why it's working. I tried changing various things in Istio model pom and Makefile but none of them seem to fix the issue. 

During model generation process, only during Istio model compilation maven-compiler-plugin throws these warnings:
```
[INFO] Artifact org.junit:junit-bom:pom:5.10.1 is present in the local repository, but cached from a remote repository ID that is unavailable in current build context, ver
ifying that is downloadable from [central (https://repo.maven.apache.org/maven2, default, releases), plexus.snapshots (https://oss.sonatype.org/content/repositories/plexus
-snapshots, default, snapshots), apache.snapshots (https://repository.apache.org/snapshots, default, snapshots)]
```

and fails with this error:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile (default-compile) on project istio-model-v1alpha3: Execution default-compile of goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile failed: Plugin org.apache.maven.plugins:maven-compiler-plugin:3.12.1 or one of its dependencies could not be resolved: Failed to collect dependencies at org.apache.maven.plugins:maven-compiler-plugin:jar:3.12.1 -> org.codehaus.plexus:plexus-compiler-api:jar:2.14.2: Failed to read artifact descriptor for org.codehaus.plexus:plexus-compiler-api:jar:2.14.2: Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact org.junit:junit-bom:pom:5.10.1 has not been downloaded from it before.
```

Following the error message, I saw that Junit5 bom is added in [plexus-compiler-api/pom.xml](https://github.com/codehaus-plexus/plexus-compiler/blob/master/pom.xml#L99) .  When I added same bom in our pom.xml, it started working.

Reply:
We should try to align versions with plexus-compiler-api then, or try to figure out the dependency convergence issue.